HLD pipeline fails with no output on the Transform fabrikate definitions task

**Describe the bug:**
From the [docs](https://github.com/microsoft/bedrock/blob/master/docs/hld-to-manifest.md#deploy-manifest-generation-pipeline) I ran
```
 spk hld install-manifest-pipeline
```

and ended up with errors on the build:

![image](https://user-images.githubusercontent.com/648372/79416479-a3d00e80-7f64-11ea-88ae-4e4365d620e2.png)

Text of errors:

```
##[error]Could not find a part of the path '/home/vsts/agents/2.166.2/_diag/pages/48028dce-11c6-43f6-8dae-15ffc6036872_1.log'. 
,##[error]Error reported in diagnostic logs. Please examine the log for more details. 
 - /home/vsts/agents/2.166.2/_diag/Worker_20200415-191306-utc.log 
System.ObjectDisposedException: The CancellationTokenSource has been disposed. 
 at System.Threading.CancellationTokenSource.ThrowObjectDisposedException() 
 at System.Threading.CancellationTokenSource.get_Token() 
 at Microsoft.VisualStudio.Services.Agent.Worker.ExecutionContext.get_CancellationToken() 
 at Microsoft.VisualStudio.Services.Agent.Worker.JobExtension.FinalizeJob(IExecutionContext jobContext) 
 at Microsoft.VisualStudio.Services.Agent.Worker.JobRunner.RunAsync(AgentJobRequestMessage message, CancellationToken jobRequestCancellationToken) 
 at Microsoft.VisualStudio.Services.Agent.Worker.JobRunner.RunAsync(AgentJobRequestMessage message, CancellationToken jobRequestCancellationToken) 
 at Microsoft.VisualStudio.Services.Agent.Worker.Worker.RunAsync(String pipeIn, String pipeOut) 
 at Microsoft.VisualStudio.Services.Agent.Worker.Program.MainAsync(IHostContext context, String[] args) 
Pool: Azure Pipelines 
Image: ubuntu-latest 
Agent: Hosted Agent 
Started: Just now 
Duration: 12s
```

```
Error analysis

Transform fabrikate definitions and publish to YAML manifests to repo (ShellScript)

Exit code 7 returned from process: file name '/home/vsts/agents/2.166.2/externals/node/bin/node', arguments '"/home/vsts/work/_tasks/ShellScript_6c731c3c-3c68-459a-a5c9-bde6e6595b5b/2.165.0/shellscript.js"'.
```


**To Reproduce:**
Run the `spk hld install-manifest-pipeline` command with a `spk.config` file with invalid `manifest git repo`:

```
azure_devops:
  org: user
  project: spk
  hld_repository: 'https://user@dev.azure.com/user/spk/_git/spk-hld'
  manifest_repository: 'https://user@dev.azure.com/user/spk/_git/spk-manifests'
```

notice the hld/manifest repos urls have `user` with `@` symbol before `dev.azure.com`

**Expected behavior:**
pipeline fails with error message and logs 

**Screenshots:**

**Additional context:**
This seems like it is a bug related to AzDo that doesn't fail properly when a git clone fails.  I will try to replicate on a minimal pipeline later.

I was able to track the issue down by running my own self hosted agent which failed with the proper error return code and logs where collected.  In the host agent it seems the timeout wipes out all the relevant logs.  If you watch the build output you can see the tasks executes but when it fails it logs are wiped clean and it seems to happen becuase a task get canceled in c#.

**Does this require updates to documentation?:**
no

--- a/gitops/azure-devops/build.sh
+++ b/gitops/azure-devops/build.sh
@@ -194,6 +194,7 @@ function git_connect() {
 
     echo "GIT CLONE: https://automated:<ACCESS_TOKEN_SECRET>@$repo_url"
     git clone "https://automated:$ACCESS_TOKEN_SECRET@$repo_url"
+    retVal=$? && [ $retVal -ne 0 ] && exit $retVal
 
     # Extract repo name from url
     repo_url=$REPO

